tests/http/server.cpp

```diff
@@ -212,18 +212,30 @@ BOOST_AUTO_TEST_CASE(registration)
     BOOST_CHECK( server.unregister( foo ));
     BOOST_CHECK( !server.unregister( foo ));
 
+    BOOST_CHECK( server.register_( "foo", [](){ return "bla"; } ));
+    BOOST_CHECK( !server.register_( "foo", [](){ return "bla"; } ));
+    BOOST_CHECK( server.unregister( "foo" ));
+    BOOST_CHECK( !server.unregister( "foo" ));
+
     BOOST_CHECK( server.subscribe( foo ));
     BOOST_CHECK( !server.subscribe( foo ));
     BOOST_CHECK( server.unsubscribe( foo ));
     BOOST_CHECK( !server.unsubscribe( foo ));
 
+    BOOST_CHECK( server.subscribe( "foo",
+                                   []( const std::string& ) { return true; }));
+    BOOST_CHECK( !server.subscribe( "foo",
+                                    []( const std::string& ) { return true; }));
```

Comment:
Late to change these names, but since the http server does not implement pub/sub communication, and I always have to look into the documentation to find out which callback setter is for PUT and which one is for GET, I have the impression that these functions should have been named differently.